fission/src/ui/modals/mirabuf/ImportMirabufModal.tsx

```diff
@@ -109,55 +171,33 @@ const ImportMirabufModal: React.FC<ModalPropsImpl> = ({ modalId }) => {
-                                ? ItemCard({
-                                      name: `DIR: ${x.displayName}`,
-                                      id: x.id,
-                                      buttonText: ">",
-                                      onClick: () => setSelectedFolder(x),
-                                  })
-                                : x instanceof Item
-                                  ? ItemCard({
-                                        name: `${x.displayName}`,
-                                        id: x.id,
-                                        buttonText: "import",
-                                        onClick: () => {
-                                            console.log(`Selecting ${x.displayName} (${x.id})`)
-                                        },
-                                    })
-                                  : ItemCard({
-                                        name: `${x.type}: ${x.id}`,
-                                        id: x.id,
-                                        buttonText: "---",
-                                        onClick: () => {
-                                            console.log(`Selecting (${x.id})`)
-                                        },
-                                    })
-                        )}
+                <Label size={LabelSize.Medium} className="text-center border-b-[1pt] mt-[4pt] mb-[2pt] mx-[5%]">
+                    {cachedRobotElements ? `${cachedRobotElements.length} Saved Robots${cachedRobotElements.length > 1 ? 's' : ''}` : "No Saved Robots"}
+                </Label>
+                {cachedRobotElements}
+                <Label size={LabelSize.Medium} className="text-center border-b-[1pt] mt-[4pt] mb-[2pt] mx-[5%]">
+                    {cachedFieldElements ? `${cachedFieldElements.length} Saved Fields${cachedFieldElements.length > 1 ? 's' : ''}` : "No Saved Fields"}
+                </Label>
+                {cachedFieldElements}
+                <Label size={LabelSize.Medium} className="text-center border-b-[1pt] mt-[4pt] mb-[2pt] mx-[5%]">
+                    {hubElements ? `${hubElements.length} Remote Asset${hubElements.length > 1 ? 's' : ''}` : "No Remote Assets"}
+                </Label>
+                {hubElements}
+                <Label size={LabelSize.Medium} className="text-center border-b-[1pt] mt-[4pt] mb-[2pt] mx-[5%]">
+                    {remoteRobotElements ? `${remoteRobotElements.length} Default Robots${remoteRobotElements.length > 1 ? 's' : ''}` : "No Default Robots"}
+                </Label>
+                {remoteRobotElements}
+                <Label size={LabelSize.Medium} className="text-center border-b-[1pt] mt-[4pt] mb-[2pt] mx-[5%]">
+                    {remoteFieldElements ? `${remoteFieldElements.length} Default Field${remoteFieldElements.length > 1 ? 's' : ''}` : "No Default Fields"}
+                </Label>
+                {remoteFieldElements}
```

Comment:
For the labels, could you differentiate between `undefined` and an empty array to indicating loading?